clients/unshaded/src/test/java/tachyon/client/block/BlockStoreContextTest.java

```diff
@@ -75,11 +83,77 @@ public void acquireAtMaxLimitTest() throws Exception {
     }
   }
 
-  class AcquireClient implements Runnable {
+  class AcquireMasterClient implements Runnable {
     @Override
     public void run() {
       BlockMasterClient client = BlockStoreContext.INSTANCE.acquireMasterClient();
       BlockStoreContext.INSTANCE.releaseMasterClient(client);
     }
   }
+
+  /**
+   * This test ensures acquiring all the available BlockStore worker clients blocks further requests
+   * for clients. It also ensures clients are available for reuse after they are released by the
+   * previous owners. If the test takes longer than 10 seconds, a deadlock most likely occurred
+   * preventing the release of the worker clients.
+   *
+   * @throws Exception if an unexpected error occurs during the test
+   */
+  @Test(timeout = 10000)
+  public void acquireWorkerLimitTest() throws Exception {
+    // Use mocks for the block worker client to prevent it from trying to invoke the session
+    // heartbeat RPC.
+    BlockWorkerClient mock = PowerMockito.mock(BlockWorkerClient.class);
+    PowerMockito.doNothing().when(mock).sessionHeartbeat();
```

Comment:
doing nothing and returning null is the default behavior for mocks' methods, so this line is unnecessary